tlsconfig/testdata/generator.go

```diff
@@ -13,69 +13,156 @@ import (
 	"fmt"
 	"io/ioutil"
 	"math/big"
+	"net"
+	"time"
 )
 
 const (
-	cert1File         = "cert-1.pem"
-	key1File          = "key-1.pem"
-	cert2File         = "cert-2.pem"
+	caCertFile     = "ca-cert.pem"
+	caKeyFile      = "ca-key.pem"
+	serverCertFile = "server-cert.pem"
+	serverKeyFile  = "server-key.pem"
+	clientCertFile = "client-cert.pem"
+	clientKeyFile  = "client-key.pem"
+
 	combinedCertsFile = "combined-certs.pem"
 	certWithKeyFile   = "cert-with-key.pem"
 )
 
+var (
+	serverTemplate = x509.Certificate{
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(10 * 365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		BasicConstraintsValid: true,
+		DNSNames:              []string{"localhost"},
+		IPAddresses:           []net.IP{net.IPv4(127, 0, 0, 1)},
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+	}
+	clientTemplate = x509.Certificate{
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(10 * 365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		BasicConstraintsValid: true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+	}
+)
+
 func main() {
-	cert1, key1 := newTestKeyPair(1, "test", "localhost")
-	err := ioutil.WriteFile(cert1File, []byte(cert1), 0644)
+	caCertStr, caKeyStr := newCAKeyPair(1, "Root CA")
+	block, _ := pem.Decode([]byte(caCertStr))
+	if block == nil {
+		panic("failed to parse certificate PEM")
+	}
+	caCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		panic(err)
+	}
+
+	serverCertStr, serverKeyStr := newSignedKeyPair(2, "Test Org", "localhost", &serverTemplate, caCert, caKeyStr)
+	clientCertStr, clientKeyStr := newSignedKeyPair(3, "Test Org", "client", &clientTemplate, caCert, caKeyStr)
+
+	err = ioutil.WriteFile(caCertFile, []byte(caCertStr), 0644)
 	if err != nil {
 		panic(err)
 	}
-	err = ioutil.WriteFile(key1File, []byte(key1), 0644)
+	err = ioutil.WriteFile(caKeyFile, []byte(caKeyStr), 0644)
+	if err != nil {
+		panic(err)
+	}
+
+	err = ioutil.WriteFile(serverCertFile, []byte(serverCertStr), 0644)
+	if err != nil {
+		panic(err)
+	}
+	err = ioutil.WriteFile(serverKeyFile, []byte(serverKeyStr), 0644)
+	if err != nil {
+		panic(err)
+	}
+
+	err = ioutil.WriteFile(clientCertFile, []byte(clientCertStr), 0644)
 	if err != nil {
 		panic(err)
 	}
-	cert2, _ := newTestKeyPair(2, "test", "test.com")
-	err = ioutil.WriteFile(cert2File, []byte(cert2), 0644)
+	err = ioutil.WriteFile(clientKeyFile, []byte(clientKeyStr), 0644)
 	if err != nil {
 		panic(err)
 	}
-	combinedCerts := fmt.Sprintf("%s%s", cert1, cert2)
+
+	combinedCerts := fmt.Sprintf("%s%s", caCertStr, serverCertStr)
```

Comment:
Consider `fmt.Sprint` or `caCertStr+serverCertStr` since this is simple string concatenation